addons/crm/tests/test_crm_lead.py

```diff
@@ -250,13 +250,13 @@ def test_crm_lead_currency_sync(self):
             'name': 'Lead 1',
             'company_id': self.company_main.id
         })
-        self.assertEqual(lead.company_currency, self.env.ref('base.EUR'))
+        self.assertEqual(lead.company_currency, self.env.ref('base.USD'))
```

Comment:
Always wondering if it is better to compare to an hardcoded value (EUR/USD) or compare to company_main.currency_id (which would nullifies the test if it is CHF, as the onchange-and-keep like behavior would not be tested anymore).

Reply:
In my opinion it looks better to compare the hard-coded value (EUR/USD).

However, I think we can replace the assertion checking the current currency of the company by setting the company currency to the hard-coded value.

Then we can remove the part setting back the currency to the original company currency, as the changes are not committed anyway.

I amended my commit with the above changes.